pkg/controller/sandbox/core/common_control.go

```diff
@@ -193,3 +234,70 @@ func (r *commonControl) createPod(ctx context.Context, box *agentsv1alpha1.Sandb
 	logger.Info("Create pod success", "Body", utils.DumpJson(pod))
 	return pod, nil
 }
+
+func (r *commonControl) handleInplaceUpdateSandbox(ctx context.Context, args EnsureFuncArgs) (bool, error) {
+	pod, box, newStatus := args.Pod, args.Box, args.NewStatus
+	logger := logf.FromContext(ctx).WithValues("sandbox", klog.KObj(box))
+
+	_, hashWithoutImageAndResource := HashSandbox(box)
+	// old Pod do not include Labels[pod-template-hash] and do not support inplace update.
+	if pod.Labels[agentsv1alpha1.PodLabelTemplateHash] == "" {
+		return true, nil
+	} else if box.Annotations[agentsv1alpha1.SandboxHashWithoutImageAndResources] != hashWithoutImageAndResource {
+		logger.Info("sandbox hash-without-image-resources changed, and does not permit in-place upgrades", "old hash",
+			box.Annotations[agentsv1alpha1.SandboxHashWithoutImageAndResources], "new hash", hashWithoutImageAndResource)
+		r.recorder.Eventf(box, corev1.EventTypeWarning, "InplaceUpdateForbidden", "InplaceUpdate only support image, resources")
+		return true, nil
+	}
+	// revision consistent
+	if pod.Labels[agentsv1alpha1.PodLabelTemplateHash] == newStatus.LatestRevision {
+		// inplace update is incompleted
+		if !inplaceupdate.IsInplaceUpdateCompleted(pod) {
+			return false, nil
+		}
+		cond := metav1.Condition{
+			Type:               string(agentsv1alpha1.SandboxConditionInplaceUpdate),
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: metav1.Now(),
+		}
+		utils.SetSandboxCondition(newStatus, cond)
+		return true, nil
+	}
+
+	state, err := inplaceupdate.GetPodInPlaceUpdateState(pod)
+	if err != nil {
+		return false, err
+		// state!=nil indicates that an in-place upgrade has already been performed previously.
+	} else if state != nil {
+		// currently, multiple in-place updates are not currently supported.
```

Comment:
// currently, multiple in-place updates are not currently supported.
-> 
// currently, multiple in-place updates are not supported.